application/src/main/java/org/thingsboard/server/service/security/system/DefaultSystemSecurityService.java

```diff
@@ -278,38 +279,38 @@ public void logLoginAction(User user, Object authenticationDetails, ActionType a
                         browser += " " + userAgent.userAgent.major;
                         if (userAgent.userAgent.minor != null) {
                             browser += "." + userAgent.userAgent.minor;
-                                if (userAgent.userAgent.patch != null) {
-                                    browser += "." + userAgent.userAgent.patch;
-                                }
+                            if (userAgent.userAgent.patch != null) {
+                                browser += "." + userAgent.userAgent.patch;
                             }
                         }
                     }
-                    if (userAgent.os != null) {
-                        os = userAgent.os.family;
-                        if (userAgent.os.major != null) {
-                            os += " " + userAgent.os.major;
-                            if (userAgent.os.minor != null) {
-                                os += "." + userAgent.os.minor;
-                                if (userAgent.os.patch != null) {
-                                    os += "." + userAgent.os.patch;
-                                    if (userAgent.os.patchMinor != null) {
-                                        os += "." + userAgent.os.patchMinor;
-                                    }
+                }
+                if (userAgent.os != null) {
+                    os = userAgent.os.family;
+                    if (userAgent.os.major != null) {
+                        os += " " + userAgent.os.major;
+                        if (userAgent.os.minor != null) {
+                            os += "." + userAgent.os.minor;
+                            if (userAgent.os.patch != null) {
+                                os += "." + userAgent.os.patch;
+                                if (userAgent.os.patchMinor != null) {
+                                    os += "." + userAgent.os.patchMinor;
                                 }
                             }
                         }
                     }
-                    if (userAgent.device != null) {
-                        device = userAgent.device.family;
-                    }
+                }
+                if (userAgent.device != null) {
+                    device = userAgent.device.family;
                 }
             }
+        }
         if (actionType == ActionType.LOGIN && e == null) {
             userService.setLastLoginTs(user.getTenantId(), user.getId());
         }
         auditLogService.logEntityAction(
                 user.getTenantId(), user.getCustomerId(), user.getId(),
-                user.getName(), user.getId(), null, actionType, e, clientAddress, browser, os, device);
+                user.getName(), user.getId(), null, actionType, e, clientAddress, browser, os, device, user.getAdditionalInfo());
```

Comment:
to be in sync with the current implementation, you'll need to fetch the provider before this line and pass to logEntityAction method 'provider' variable, and not the full additional info object of the user.
Plus, there is mixing naming - additonalInfo as JSON from user and additonalInfo as Object... array.